packages/runtime/src/next/root-style-registry.tsx

```diff
@@ -7,26 +7,32 @@ import { ReactNode, createContext, useContext, useState } from 'react'
 
 const CacheContext = createContext(cache)
 
-export function RootStyleRegistry({ children }: { children: ReactNode }) {
-  const [{ cache, flush }] = useState(() => {
-    const cache = createCache({ key: 'css' })
-    cache.compat = true
-    const prevInsert = cache.insert
-    let inserted: string[] = []
-    cache.insert = (...args) => {
-      const serialized = args[1]
-      if (cache.inserted[serialized.name] === undefined) {
-        inserted.push(serialized.name)
-      }
-      return prevInsert(...args)
-    }
-    const flush = () => {
-      const prevInserted = inserted
-      inserted = []
-      return prevInserted
+const createRootStyleCache = () => {
```

Comment:
(Toggle on the "Hide whitespace" setting for an easier diff).

Factored out `createRootStyleCache` from inside of the `useState` hook/added some newlines for easier comprehension as I was debugging this, no semantic changes.